packages/app-builder-lib/src/targets/nsis/nsisUtil.ts

```diff
@@ -39,30 +39,44 @@ export const NSIS_PATH = () => {
   })
 }
 
+export interface PackArchResult {
+  fileInfo: PackageFileInfo
+  unpackedSize: number
+}
+
 export class AppPackageHelper {
-  private readonly archToFileInfo = new Map<Arch, Promise<PackageFileInfo>>()
+  private readonly archToResult = new Map<Arch, Promise<PackArchResult>>()
   private readonly infoToIsDelete = new Map<PackageFileInfo, boolean>()
 
   /** @private */
   refCount = 0
 
   constructor(private readonly elevateHelper: CopyElevateHelper) {}
 
-  async packArch(arch: Arch, target: NsisTarget): Promise<PackageFileInfo> {
-    let infoPromise = this.archToFileInfo.get(arch)
-    if (infoPromise == null) {
+  async packArch(arch: Arch, target: NsisTarget): Promise<PackArchResult> {
+    let resultPromise = this.archToResult.get(arch)
+    if (resultPromise == null) {
       const appOutDir = target.archs.get(arch)!
-      infoPromise = this.elevateHelper.copy(appOutDir, target).then(() => target.buildAppPackage(appOutDir, arch))
-      this.archToFileInfo.set(arch, infoPromise)
+      resultPromise = this.elevateHelper
+        .copy(appOutDir, target)
+        .then(() => dirSize(appOutDir))
+        .then(unpackedSize =>
+          target.buildAppPackage(appOutDir, arch).then(fileInfo => ({
+            fileInfo,
+            unpackedSize,
+          }))
```

Comment:
`buildAppPackage` seems to modify the `appOutDir` again, after `dirSize` has been called. (Also can't guarantee if that functionality will be changed in the future.)

What are your thoughts on this? It feels more future-proof
```
      resultPromise = this.elevateHelper
        .copy(appOutDir, target)
        .then(() => target.buildAppPackage(appOutDir, arch))
        .then(async fileInfo => ({
            fileInfo,
            unpackedSize: await dirSize(appOutDir),
          }))
```

Reply:
Done, thanks